src/Core/Generator/SchemaGeneratorFactory.cs

```diff
@@ -56,46 +58,78 @@ public static async Task<string> Create(RuntimeConfig config, string mode, int?
             }
 
             string? connectionString = config.DataSource?.ConnectionString;
-            string? databaseName = config.DataSource?.Options?["database"]?.ToString();
-            string? containerName = config.DataSource?.Options?["container"]?.ToString();
 
-            if (connectionString == null || databaseName == null || containerName == null)
+            object? globalDatabaseName = null;
+            object? globalContainerName = null;
+
+            config.DataSource?.Options?.TryGetValue("database", out globalDatabaseName);
+            config.DataSource?.Options?.TryGetValue("container", out globalContainerName);
+
+            if (string.IsNullOrEmpty(connectionString) || globalDatabaseName is null)
             {
-                logger.LogError("Connection String, Database and container name must be provided in the config file");
-                throw new ArgumentException("Connection String, Database and container name must be provided in the config file");
+                logger.LogError("Connection String and Database name must be provided in the config file");
+                throw new ArgumentException("Connection String and Database name must be provided in the config file");
             }
 
-            logger.LogInformation("Connecting to Cosmos DB Database: {0}, Container: {1}", databaseName, containerName);
+            HashSet<string> dbAndContainerToProcess = new();
+            if (globalDatabaseName != null && globalContainerName != null)
```

Comment:
if globalDatabaseName is provided but globalContainerName is not, what is your course of action?

Reply:
it will check at entity level, what all containers are mentioned, and it will scan only those containers for schema generation.